kategory-core/src/test/kotlin/kategory/data/OptionTest.kt

```diff
@@ -60,6 +60,31 @@ class OptionTest : UnitSpec() {
             Option.fromNullable(a) shouldBe None
         }
 
+        "toOption should work for both null and non-null values of nullable types" {
+            forAll { a: Int? ->
+                // This seems to be generating only non-null values, so it is complemented by the next test
+                val o: Option<Int> = a.toOption()
+                if (a == null) o == None else o == Some(a)
+            }
+        }
+
+        "toOption should return none for null values of nullable types" {
+            val a: Int? = null
+            a.toOption() shouldBe None
+        }
+
+        "toNullable should return null for none values" {
+            val a: Option<Int> = Option.none()
+            a.toNullable() shouldBe null
+        }
+
+        "toNullable should work for both null and non-null values of nullable types" {
+            forAll { a: Int? ->
+                // This seems to be generating only non-null values, so it is complemented by the next test
+                val o: Option<Int> = Option.some(a)
```

Comment:
The compiler is complaining about this call: https://travis-ci.org/kategory/kategory/builds/296163693

```
e: /home/travis/build/kategory/kategory/kategory-core/src/test/kotlin/kategory/data/OptionTest.kt: (84, 45): Type inference failed. Expected type mismatch: inferred type is Option<Option.Companion> but Option<Int> was expected
e: /home/travis/build/kategory/kategory/kategory-core/src/test/kotlin/kategory/data/OptionTest.kt: (84, 50): Too many arguments for public fun <A> A.some(): Option<A> defined in kategory
```

Reply:
You can run the tests locally using `./gradlew test`

Reply:
The cause is that `Some` is the constructor and `some` is an extension function to make the value into a Some. You're making the companion object into a Some, instead of a :D